src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -852,32 +852,56 @@ internal unsafe int GetHashCodeOfStringCore(ReadOnlySpan<char> source, CompareOp
                 return 0;
             }
 
-            fixed (char* pSource = source)
+            // according to ICU User Guide: 
+            // "The overall performance is poorer than if the function is called with a zero output buffer"
+            // "A recommended size for the temporary buffer is four times the length of the longest string processed"
+            int recommendedSize = source.Length * 4; 
```

Comment:
Potential integer overflow.

Reply:
Good point, for some reason I always forget about it